src/store/selectors/subjectsPicker.js

```diff
@@ -1,53 +1,19 @@
 const subjectsPicker = ({ primary, lingual }) => {
-  const maxScoreWrapper = {
-    primary: {},
-    lingual: {}
-  }
-
-  let maxPrimaryScore = 0
-  let maxLingualScore = 0
-  let maxPrimarySubject
-  let maxLingualSubject
-  let emptySubject = {
-    name: '',
-    max: {
-      score: 0,
-      formula: ''
-    }
-  }
-
-  maxPrimarySubject = maxLingualSubject = emptySubject
+  const maxScoreWrapper = {}
```

Comment:
That's much better to what was before.
I have played with this a little bit (too much ;) ) and came up with following one.
You do not need to replace your solution with mine, however It contains many small changes and optimizations, some of them you might find useful.
```
const subjectsPicker = ({ primary, lingual }) => {
  const [bestPrimary] = _sortSubjectsByScore(primary)
  const [bestLingual, secondBestLingual] = _sortSubjectsByScore(lingual)

  return lingualAsPrimaryScore(bestLingual, bestPrimary, secondBestLingual) ||
    { primary: bestPrimary, lingual: bestLingual }
}

const _sortSubjectsByScore = (subjects) => {
  return subjects.sort((subjectA, subjectB) => {
    if (subjectA.max.score > subjectB.max.score) return -1
    if (subjectA.max.score < subjectB.max.score) return 1
    return 0
  })
}

const lingualAsPrimaryScore = (bestLingual, bestPrimary, secondBestLingual) => {
  if (bestLingual.max.score * 4 <= bestPrimary.max.score) return

  const totalBest = _getTotalBest(bestPrimary, bestLingual)
  const lingualTotalBest = _getTotalBest(bestLingual, secondBestLingual)

  if (lingualTotalBest > totalBest) {
    return { primary: bestLingual, lingual: secondBestLingual }
  }
}

const _getTotalBest = (bestPrimary, bestLingual) => bestPrimary.max.score * 4 + bestLingual.max.score

export default subjectsPicker
```

Most of improvements from above could be applied also for getCourseScore.